servicebroker/lib/routes/binding.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+
+module.exports = function(app) {
+  var path = require('path');
+  var fs = require('fs');
+  var request = require('request');
+  var logger = require(path.join(__dirname, '../logger/logger.js'));
+  var models = require(path.join(__dirname, '../models'))();
+
+  var settings = JSON.parse(
+    fs.readFileSync(path.join(__dirname, '../../config/settings.json'), 'utf8'));
+  var apiServerUtil = require(path.join(__dirname, '../util/apiServerUtil.js'))(settings.apiServerUri);
+  var messageUtil = require(path.join(__dirname, '../util/messageUtil.js'))();
+  app.put('/v2/service_instances/:instance_id/service_bindings/:binding_id', function(req, res) {
+    var serviceInstanceId = req.params.instance_id;
+    var bindingId = req.params.binding_id;
+    var appId = req.body.app_guid;
+    var policyJSON = req.body.parameters;
+    if (typeof(policyJSON) === "undefined") {
+      res.status(400).json({ "description": messageUtil.getMessage("POLICY_REQUIRED") });
+    } else {
+      models.binding.findAll({ where: { appId: appId } }).then(function(result) {
+        var length = result.length;
+        if (length === 0) {//no binding , so create one
+          models.sequelize.transaction().then(function(t) {
```

Comment:
Shouldn't the findAll be part of the transaction? Otherwise I can have 2 requests that both fail the findAll.